puts/harness/include/tlspuffin/put.h

```diff
@@ -0,0 +1,155 @@
+#ifndef TLSPUFFIN_PUT_H
+#define TLSPUFFIN_PUT_H
+
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+// FIXME: C/Rust duplication of type definitions
+//
+//     All the types related to the descriptor are equivalent of the Rust
+//     structs and enums. To avoid discrepancy, we could generate a header from
+//     Rust using `cbindgen` or use the C definition in Rust using `bindgen`.
+//
+//     Another solution would be to expose functions to retrieve config
+//     informations from Rust. This has the advantage to completely hide the
+//     rust structs at the price of a small performance cost each time one of
+//     these functions is called.
+
+typedef void *RESULT;
+
+typedef enum
+{
+    V1_3,
+    V1_2
+} TLS_VERSION;
+
+typedef enum
+{
+    CLIENT,
+    SERVER
+} AGENT_TYPE;
+
+typedef struct
+{
+    const uint8_t *const bytes;
+    const size_t length;
+} PEM;
+
+typedef enum
+{
+    RESULT_OK,
+    RESULT_IO_WOULD_BLOCK,
+    RESULT_ERROR_FATAL,
+    RESULT_ERROR_OTHER
+} RESULT_CODE;
+
+typedef struct
+{
+    uint8_t name;
+    AGENT_TYPE type;
+    TLS_VERSION tls_version;
+    bool client_authentication;
+    bool server_authentication;
+    const PEM *cert;
+    const PEM *pkey;
+
+    const PEM *const *const store;
+    const size_t store_length;
+} AGENT_DESCRIPTOR;
+
+typedef struct C_HARNESS_TYPE
+{
+    const char *name;            // openssl
+    const char *version;         // tlspuffin v0.1.0 (commit f80370)
+} C_HARNESS_TYPE;
+
+typedef struct C_LIBRARY_TYPE
+{
+    const char *vendor_name;     // openssl
+    const char *vendor_version;  // OpenSSL 1.1.1j 16 Feb 2021
+
+    const char *config_name;     // openssl111j
+    const char *config_hash;     // da39e788eca8dabb
```

Comment:
What is a config hash? Do we need to store this field or can we derive from config_name?

Reply:
The config hash corresponds to the vendor library build configuration. In this PR it is based on the path to the provided library but in the future it will match the configuration selected through `mk_vendor`. The idea is to have a unique identifier for the vendor library, so that we are able to distinguish between different builds.

For now, `mk_vendor` includes the following information in the hash:
- the fetch configuration for the sources (e.g. git url + git ref)
- a hash of the patches applied on these sources
- a hash of the wrapper build script used to build these sources (could be replaced with the version of the tlspuffin workspace once we use a release process enforcing semver)
- the build parameters passed to the wrapper build script (WITH_ASAN, WITH_GCOV, ...)

Reply:
Forgot to answer the actual question 😅 : We cannot derive it from the name or version.